consensus/src/dag/dag_fetcher.rs

```diff
@@ -91,8 +91,22 @@ impl DagFetcher {
                     local_request.notify();
                     continue;
                 }
+
+                let target = local_request.node();
                 RemoteFetchRequest::new(
-                    local_request.node().metadata().clone(),
+                    target.metadata().epoch(),
+                    local_request
+                        .node()
+                        .parents()
+                        .iter()
+                        .filter_map(|node| {
+                            if dag_reader.get_node(node.metadata()).is_some() {
+                                None
+                            } else {
+                                Some(node.metadata().clone())
+                            }
+                        })
```

Comment:
Can you merge the all_exists() check (from above) here?

Reply:
we probably should just return the missing ones from `all_exists` function?

Reply:
all_exists is used in many places and doesn't need to clone nodemetadata everywhere. I can introduce a new fn.